src/components/Navigation/NavigationItem/NavigationItem.tsx

```diff
@@ -55,15 +61,22 @@ const NavigationItem = forwardRef<HTMLLIElement, NavigationItemProps>(
       navGroup,
       navType,
       onClick,
+      closeSubmenu,
       element: Component = Link,
       ...props
     },
     ref,
   ) => {
+    const { closeMenu } = React.useContext(HeaderContext);
+
     return (
       <li
         {...props}
-        onClick={onClick}
+        onClick={(e) => {
+          closeMenu?.();
+          closeSubmenu?.();
+          onClick?.(e);
```

Comment:
If we just need to pass in the state update function, is it possible to leverage the closeMenu or onClick handlers?

Reply:
Can't we just pass in whatever callback we want, one that closes all menus. So if they are all state setters
```js
onCloseMenu = () => {
  setIsMobileMenuOpen(false);
  setIsDropdownExpanded(false);
 }
```
then in the component we can just call `onCloseMenu` on click?

Reply:
Just wondering if we need to increase our Prop API or if we can just leverage the onClick that is already applied.